erpnext/stock/doctype/pick_list/pick_list.py

```diff
@@ -41,8 +41,15 @@ def before_save(self):
 	def before_submit(self):
 		update_sales_orders = set()
 		for item in self.locations:
-			# if the user has not entered any picked qty, set it to stock_qty, before submit
-			if item.picked_qty == 0:
+			if self.scan_mode and item.picked_qty < item.stock_qty:
+				frappe.throw(
+					_("Row {0} is short by {1} {2}").format(
```

Comment:
```suggestion
					_("Row {0} is short by {1} {2}").format(
```

Better wording than "short by" 😅 

Perheps: `Row {} picked qty is less than required qty, additional {pending_qty} {uom} required.`